### PR TITLE
Feature/issue 150  Add Event V0.5 feature

### PR DESCRIPTION
**Pull request by lduf**
### Description

**Event**
User can : create, update, view, viewAny, delete (CRUD) events
Event is considered as a list

- Create some artisan command to handle theses new feature.
- Debug some cytoscape error (issues with role)
- Debug Create link on a new team (no more blank fields)

**Add the ability to add multiple relations :** 
- Add from/to every person in the same event

**Auto change team :** 
When someone creates a new team, he is automatically associated to the newly created team.

⚠️ DO NOT rollback

Please run the following : 
```
php artisan migrate
php artisan migrate:refresh --path=database/migrations/2022_03_07_160104_create_events_table.php
php artisan permissions:update
php artisan  lists:update --seed
```

## Issues fixed
List of related issues that are fixed by this PR :

see Development

### Checklist
- [ ] Compiling
- [ ] Code factoring (comments, indent...)
- [ ] Documentation (net4p Doc, Readme...)

### Feature Checklist
- [ ] Cytoscape
- [ ] Person/Index
- [ ] Person/Filter
- [ ] Person/Create
- [ ] Person/Edit
- [ ] Person/Delete
- [ ] Relation/Index
- [ ] Relation/Filter
- [ ] Relation/Create
- [ ] Relation/Edit
- [ ] Relation/Delete
- [ ] Duplicates/Index
- [ ] Duplicates/Filter
- [ ] Duplicates/Create
- [ ] Duplicates/Edit
- [ ] ManageList/Delete
- [ ] ManageList/Create
- [ ] ManageList/Edit
- [ ] ManageField/Index
- [ ] ManageField/Create
- [ ] ManageField/Edit
- [ ] ManageField/Delete
- [ ] ApiList
- [ ] Crews
- [ ] Users

### CODE REVIEW

**Comment by lduf** [comment]
@mathisfchx  please try the following features (check the checkbox if it's working): 

- [ ] create a new team
- [ ] create a new form (with the event list field)
- [ ] create a new event
- [ ] add several person 
- [ ] add relation using : add everyone from the same event (on from or on to)
- [ ] add another person
- [ ] add relation (classic way : you choose 2 people)

File: src/database/migrations/jsonDataset/roles.json [hunk 1]
```diff
@@ -40,19 +40,30 @@
             "duplicate.show",
             "duplicate.edit",
             "duplicate.update",
-            "duplicate.destroy"
+            "duplicate.destroy",
+            "event.index",
+            "event.show",
```
Comment (mathisfchx) [inline src/database/migrations/jsonDataset/roles.json:45]:
```suggestion
            
```

File: src/database/migrations/jsonDataset/roles.json [hunk 2]
```diff
@@ -40,19 +40,30 @@
             "duplicate.show",
             "duplicate.edit",
             "duplicate.update",
-            "duplicate.destroy"
+            "duplicate.destroy",
+            "event.index",
```
Comment (mathisfchx) [inline src/database/migrations/jsonDataset/roles.json:44]:
```suggestion

```

File: src/app/Http/Controllers/LinkController.php [hunk 3]
```diff
@@ -74,8 +74,33 @@ public function store(StoreLinkRequest $request)
...
+
+        if (isset($associatedPersonsThroughEvent) && !empty($associatedPersonsThroughEvent) && isset($direction)) {
+            foreach ($associatedPersonsThroughEvent as $associatedPerson) {
+                $link[$direction] = $associatedPerson->id;
+                Link::create($link);
+            }
+            return redirect()->route("links.index");
+        }
+
         Link::create($link);
```
Comment (mathisfchx) [inline src/app/Http/Controllers/LinkController.php:104]: validation failed if no event = bad

File: src/app/Models/ListControl.php [hunk 4]
```diff
@@ -162,7 +162,23 @@ public static function getAPIContent(User $user)
...
 
-    public function getDisplayedValueAttribute(){
+    public function getDisplayedValueAttribute()
+    {
         return $this->structure()->find($this->attributes['displayed_value'])->field ?? $this->attributes['displayed_value'];
     }
+
+    public static function list()
+    {
+        return self::orderBy('short')->pluck('short', 'id');
```
Comment (mathisfchx) [inline src/app/Models/ListControl.php:172]: ??

File: src/resources/views/event/create.blade.php [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+                            @error($form_elem)
+                            <p class="text-sm text-red-600">{{ $message }}</p>
+                            @enderror
+                        </div>
+
+
+                        <!--  Country SECTION  -->
+                        <div class="px-4 py-5 bg-white sm:p-6">
+                            @php($form_elem = "country_id")
+                            @php($list = \App\Models\ListCountry::list())
```
Comment (mathisfchx) [inline src/resources/views/event/create.blade.php:54]: ::list() in src/app/Models/ListControl.php = to generalize

File: src/resources/views/event/create.blade.php [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+                            <input value="{{ old($form_elem)}}" type="date" name="{{$form_elem}}" id="{{$form_elem}}"
+                                   class="form-input rounded-md shadow-sm mt-1 block w-full"/>
+                            <small id="{{$form_elem}}Help" class="block font-medium text-sm text-gray-500">The start
+                                date</small>
+                            @error($form_elem)
+                            <p class="text-sm text-red-600">{{ $message }}</p>
+                            @enderror
+                        </div>
+
+                        <!--  start date SECTION  -->
```
Comment (mathisfchx) [inline src/resources/views/event/create.blade.php:96]: section are not corresponding (also check the others)

File: src/resources/views/event/edit.blade.php [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+                            @error($form_elem)
+                            <p class="text-sm text-red-600">{{ $message }}</p>
+                            @enderror
+                        </div>
+
+
+                        <!--  Country SECTION  -->
+                        <div class="px-4 py-5 bg-white sm:p-6">
+                            @php($form_elem = "country_id")
+                            @php($list = \App\Models\ListCountry::list())
```
Comment (mathisfchx) [inline src/resources/views/event/edit.blade.php:56]: same thing (generalize)

File: src/resources/views/event/edit.blade.php [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+                                   id="{{$form_elem}}"
+                                   class="form-input rounded-md shadow-sm mt-1 block w-full"/>
+                            <small id="{{$form_elem}}Help" class="block font-medium text-sm text-gray-500">The start
+                                date</small>
+                            @error($form_elem)
+                            <p class="text-sm text-red-600">{{ $message }}</p>
+                            @enderror
+                        </div>
+
+                        <!--  start date SECTION  -->
```
Comment (mathisfchx) [inline src/resources/views/event/edit.blade.php:100]: checks section names

**Review by mathisfchx — CHANGES_REQUESTED** [review]

File: src/database/migrations/jsonDataset/roles.json (hunk 1 above)
Comment (lduf) [reply src/database/migrations/jsonDataset/roles.json:45]: done

File: src/database/migrations/jsonDataset/roles.json (hunk 2 above)
Comment (lduf) [reply src/database/migrations/jsonDataset/roles.json:44]: done

**Comment by mathisfchx** [comment]
⚠️ in update:list add a --seed for event type ⚠️

File: src/app/Http/Controllers/LinkController.php (hunk 3 above)
Comment (lduf) [reply src/app/Http/Controllers/LinkController.php:104]: done

**Comment by lduf** [comment]
> 

Already done 😆

**Review by mathisfchx — APPROVED** [review]
🌍

**Pull request merged**